zcash_primitives/src/transaction/builder.rs

```diff
@@ -473,40 +472,44 @@ impl<R: RngCore + CryptoRng> Builder<R> {
         tx_metadata.output_indices.resize(orig_outputs_len, 0);
 
         // Create Sapling SpendDescriptions
-        for (i, (pos, spend)) in spends.iter().enumerate() {
-            let proof_generation_key = spend.extsk.expsk.proof_generation_key(&JUBJUB);
+        if spends.len() > 0 {
+            let anchor = self.anchor.expect("anchor was set if spends were added");
```

Comment:
Thanks for fixing this! I recall noticing this issue a while back, but completely forgot to address it.